source/unix/openssl_platform_init.c

```diff
@@ -38,6 +38,7 @@ static struct aws_allocator *s_libcrypto_allocator = NULL;
  * and avoid dead-stripping
  */
 #if defined(OPENSSL_IS_AWSLC) || defined(OPENSSL_IS_BORINGSSL)
+/* TODO: MUSL LTO will strip the symbols even with the weak refs */
```

Comment:
can you expand on this comment with what we found?
i.e. that on alpine used weak ref does not seem to be enough and lto (or maybe some other optimization) still strips those symbols if it thinks they are not used